### DISCUSSION TITLE
Leaks in Blast using NvBlastExtAuthoringFindAssetConnectingBonds

### DISCUSSION THREAD

**Discussion by ghost** [NVIDIA-Omniverse/PhysX#183] [General]
Hi, 

I have some memory leaks using the blast plugin, that comes from using the NvBlastExtAuthoringFindAssetConnectingBonds method.
If I go through the log I have for the leaks, I have a call to NvBlastExtAuthoringTransformCollisionHull in the method, line 527 of NvBlastExtAuthoring.cpp. It then points me to these allocations in NvBlastExtAuthoringTransformCollisionHull (line 174 to 176 of NvBlastExtAuthoring.cpp) :
```
ret->points = SAFE_ARRAY_NEW(NvcVec3, ret->pointsCount);
ret->indices = SAFE_ARRAY_NEW(uint32_t, ret->indicesCount);
ret->polygonData = SAFE_ARRAY_NEW(HullPolygon, ret->polygonDataCount);
```
The CollisionHulls allocated by this method are stored in hullToRelease array, which then deletes all these CollisionHulls (line 574) : 
```
for (CollisionHull* hull : hullsToRelease)
{
    delete hull;
}
```
I did implement my custom CollisionHullImpl, to go with my custom ConvexMeshBuilder, but the destructor of my CollisionHullImpl does release these arrays (and I don't have leaks problems when using my ConvexMeshBuilder) : 

```
CollisionHullImpl::~CollisionHullImpl()
{
    SAFE_ARRAY_DELETE(points);
    SAFE_ARRAY_DELETE(indices);
    SAFE_ARRAY_DELETE(polygonData);
}
```
It seems the base CollisionHull struct has no destructor at all. The leaks might come from this?

**Comment by preist-nvidia** [comment]
Closing in favor of issue. https://github.com/NVIDIA-Omniverse/PhysX/issues/185
